test/controllers/managing.officer.controller.spec.ts

```diff
@@ -8,17 +9,72 @@ import request from "supertest";
 import app from "../../src/app";
 import { authentication } from "../../src/controllers";
 import { MANAGING_OFFICER_URL } from "../../src/config";
+import { getApplicationData, prepareData, setApplicationData } from '../../src/utils/application.data';
+import { APPLICATION_DATA_MOCK, MANAGING_OFFICER_OBJECT_MOCK } from '../__mocks__/session.mock';
+import { ANY_MESSAGE_ERROR, MANAGING_OFFICER_PAGE_HEADING, SERVICE_UNAVAILABLE } from '../__mocks__/text.mock';
+import { managingOfficerType } from '../../src/model';
 
 const mockAuthenticationMiddleware = authentication as jest.Mock;
 mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
 
-const MANAGING_OFFICER_PAGE_TITLE = "Tell us about the managing officer";
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockSetApplicationData = setApplicationData as jest.Mock;
+const mockPrepareData = prepareData as jest.Mock;
 
 describe("MANAGING_OFFICER controller", () => {
-  test("renders the managing officer page", async () => {
-    const resp = await request(app).get(MANAGING_OFFICER_URL);
 
-    expect(resp.status).toEqual(200);
-    expect(resp.text).toContain(MANAGING_OFFICER_PAGE_TITLE);
+  describe("GET tests", () => {
+    test("renders the managing officer page", async () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+      const resp = await request(app).get(MANAGING_OFFICER_URL);
+
```

Comment:
We can test the data values here - see the tests for corporate managing officers.